### DISCUSSION TITLE
Can you upgrade WPRIG  version in an existing project? Or, "How to keep a project alive as php versions and packages change?"

### DISCUSSION THREAD

**Discussion by kreativkod** [wprig/wprig#839] [Q&A]
Hi,
I ran into a problem just now, and would appreciate some help from the community if possible.

I have a project from a year ago or so using wprig@2.2.1 on Debian 11 with php8.0. 
I have now upgraded to Debian 12 and php8.2 and wprig@2.3 in a brand new project seems to be working fine. 
But my old project with 2.2.1 runs into a gulp-phpcs error upon npm run dev/bundle. 
I still have the Debian 11 partition available and in that dev environment the old project still works fine.

Is it possible to upgrade the wprig version in place in the existing project or is there some other procedure one should follow to carry a project forward into a newer wprig version?

I'd be grateful for all tips!

I have done npm update and npx update but to no avail.

This is the error I get:
`[18:28:58] 'php' errored after 6.2 s
[18:28:58] Error in plugin "gulp-phpcs"
Message:
    Execution of Code Sniffer Failed
Details:
    stdout: E 1 / 1 (100%)`

**Comment by kreativkod** [comment]
I tried `composer update` as well but it would not run due to my composer plugin being too new 2.30 for the requirement (^1.0).

So I tried something hacky on a non-critical project. 
I replaced `composer.json/.lock `and `packages.json/.lock` with the same files from the current @2.3 version of wp-rig. Then I did `npm run rig-init `and all the new packages were installed or updated. Of course this is just external packages and not the wprig functionality updated, so I guess I am still running the old wprig version but with current packages.

It seems to work, I can do `npm run dev`,  but I don't now if this project now has flaws 'under the hood' that I do not see right now?

And if so, what would otherwise be the proper way to keep working on a project using wp-rig over the years as packages come and go and php versions need to be updated?

**Comment by kreativkod** [comment]
FWIW, one thing that I find puzzling is that the error thrown upon `npm run dev` references `gulp-phpcs` which has not received a new update for 4 years, and php8.0 was not even released when it received its latest update. So it puzzles me that the move to a new development environment with php8.2 would cause this error.

**Comment by robruiz** [comment]
https://wprig.io/documentation/recommended-git-workflow/
